ui/app/models/job.js

```diff
@@ -84,6 +84,16 @@ export default Model.extend({
   taskGroups: fragmentArray('task-group', { defaultValue: () => [] }),
   summary: belongsTo('job-summary'),
 
+  // A job model created from the jobs list response will be lacking
+  // task groups. This is an indicator that it needs to be reloaded
+  // if task group information is important.
+  isPartial: equal('taskGroups.length', 0),
```

Comment:
My Nomad knowledge is weak: is it the case that a job always has one or more task groups?

If it's possible for a job to really have zero task groups (ignoring partial loads), this `isPartial` may be misleading. One fix might be to make `taskGroups` a nullable array value. This would allow `null` to represent the unknown/not-loaded state in a way that's distinct from no-task-groups.

(If all jobs have one or more task groups, then this is obviously not a problem 😅)